examples/extending/extending-leaflet.md

```diff
@@ -0,0 +1,445 @@
+    console.log(instance.options.height); // Outputs "1", the default
+    
+Leaflet handles the `options` property in a special way: options available for a parent class will be inherited by a children class:.
+
+    var MyCubeClass = MyBoxClass.extend({
+        options: {
+            depth: 1
+        }
+    });
+    
+    var instance = new MyCubeClass('Blue');
+    
+    console.log(instance.options.width);
+    console.log(instance.options.height);
+    console.log(instance.options.depth);
+
+
+It's quite common for child classes to run the parent's constructor, and then their own constructor. In Leaflet this is achieved using `L.Class.addInitHook()`. This method can be used to "hook" initialization functions that run right after the class' `initialize()`, for example:
+
+    MyBoxClass.addInitHook(function(){
+        this._area = this.options.width * this.options.length;
+    });
+
+That will run after `initialize()` (which calls `setOptions()`) is called - this means that `this.options` exist and is valid when the init hook runs.
+
+`addInitHook` has an alternate syntax, which uses method names and can fill method arguments in:
+
+    MyCubeClass.include({
+        _calculateVolume: function(arg1, arg2) {
+            this._volume = this.options.width * this.options.length * this.options.depth;
+        }
+    });
+    
+    MyCubeClass.addInitHook('_calculateVolume', argValue1, argValue2);
+    
+
+### Methods of the parent class
+
+Calling a method of a parent class is achieved by reaching into the prototype of the parent class and using [`Function.call(…)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call). This can be seen, for example, in the code for `L.FeatureGroup`:
+
+    L.FeatureGroup = L.LayerGroup.extend({
+    
+        addLayer: function (layer) {
+            …
+            L.LayerGroup.prototype.addLayer.call(this, layer);
+        },
+        
+        removeLayer: function (layer) {
+            …
+            L.LayerGroup.prototype.removeLayer.call(this, layer);
+        },
+    
+        …
+    });
+
+Calling the parent's constructor is done in a similar way, but using `ParentClass.prototype.initialize.call(this, …)` instead.
+    
+    
+### Factories    
+
+Most Leaflet classes have a corresponding [factory function](https://en.wikipedia.org/wiki/Factory_%28object-oriented_programming%29). A factory function has the same name as the class, but in `lowerCamelCase` instead of `UpperCamelCase`:
+    
+    function myBoxClass(name, options) {
+        return new MyBoxClass(name, options);
+    }
+    
+    
+### Naming conventions
+
+When naming classes for Leaflet plugins, please adhere to the following naming conventions:
+
+* Never expose global variables in your plugin.
+* If you have a new class, put it directly in the `L` namespace (`L.MyPlugin`).
+* If you inherit one of the existing classes, make it a sub-property (`L.TileLayer.Banana`).
+
```

Comment:
Maybe add a comment about the convention for namespacing factory functions? The convention is to use lowercase namespace for the factory, like `L.control.zoom`, rather than `L.Control.zoom`. I know I was confused about this when I started writing plugins.